template/epiTAD_script.template.R

```diff
@@ -0,0 +1,365 @@
+lad$chr <- as.numeric(lad$chr)
+lad <- lad[!is.na(lad$chr), ]
+
+
+# Prepare SNPs ------------------------------------------------------------
+
+# reactive: dat
+snps <- as.character(unlist(strsplit(snpList, ",")))
+snps <- trimws(snps)
+dat <- queryHaploreg(query = snps, ldThresh = value, ldPop = pop)
+dat$chr <- as.numeric(as.character(dat$chr))
+dat$pos_hg38 <- as.numeric(as.character(dat$pos_hg38))
+
+
+# Choose Tissues ----------------------------------------------------------
+
+etest <- unlist(strsplit(as.character(dat$eQTL), ";"))
+etest <- etest[!etest %in% c(".")]
+etest2 <- unlist(strsplit(etest, ","))
+if (length(etest2)) {
+  etest3 <- matrix(etest2, nrow = length(etest), ncol = 4, byrow = TRUE)
+  etest3 <- as.data.frame(etest3)
+  etest3 <- etest3[!duplicated(etest3$V2), ]
+  tissues_avail <- etest3$V2
+  tissue_missing <- setdiff(tissue, tissues_avail)
+  if (length(tissue_missing)) {
+    warning("Ignoring eQTLs ", paste(tissue_missing, collapse = ", "),
+            " as these were not associated with the requested SNPs")
+  }
+  tissue <- intersect(tissue, tissues_avail)
+} else {
+  message("No statistically significant eQTLs were reported with these SNPs.")
+}
+
+
+# Dat2 --------------------------------------------------------------------
+# TODO GAB: dat2 and associated function need more descriptive names
+dat2_function <- function(snpList) {
+  regulome_recode <- c(
+    "1a" = "eQTL + TF binding + matched TF motif + matched DNase Footprint + DNase peak",
+    "1b" = "eQTL + TF binding + any motif + DNase Footprint + DNase peak",
+    "1c" = "eQTL + TF binding + matched TF motif + DNase peak",
+    "1d" = "eQTL + TF binding + any motif + DNase peak",
+    "1e" = "eQTL + TF binding + matched TF motif",
+    "1f" = "eQTL + TF binding / DNase peak",
+    "2a" = "TF binding + matched TF motif + matched DNase Footprint + DNase peak",
+    "2b" = "TF binding + any motif + DNase Footprint + DNase peak",
+    "2c" = "TF binding + matched TF motif + DNase peak",
+    "3a" = "TF binding + any motif + DNase peak",
+    "3b" = "TF binding + matched TF motif",
+    "4"  = "TF binding + DNase peak",
+    "5"  = "TF binding or DNase peak"
+  )
+
+  snps <- as.character(unlist(strsplit(snpList, ",")))
+  snps <- trimws(snps)
+  x <- queryRegulome(query = snps)
+  if (nrow(x$res.table) < 1) {
+    stop("The queried SNP may not be valid. Please check your input.")
+  }
+  x <- as.data.frame(x$res.table)
+  x$score <- as.character(x$score)
+  x$score_anno <- ifelse(x$score %in% names(regulome_recode),
+                         regulome_recode[x$score],
+                         "Other")
+  x
+}
+
+dat2 <- dat2_function(snpList)
+dat2
+
+
+# In TAD or LAD -----------------------------------------------------------
+
+in_tad <- function(dat, snps, tad) {
+  dat <- dat[dat$rsID %in% snps, ]
+  snp_pos <- dat$pos_hg38
+  tad <- tad[tad$chr == max(dat$chr, na.rm = TRUE), ]
+  tad[tad$start_position <= snp_pos & tad$end_position >= snp_pos, ]
```

Comment:
@jhcreed This line (and the similar line in `in_lad()` throw warnings when `snp_pos` has length > 1 (both in script and app). When `length(snp_pos) > 1`, I think only the first element is used, so `in_tad()` only checks the first SNP. You may want to investigate to verify that this is the case or the desired behavior.